server/sonar-server/src/main/java/org/sonar/server/component/ComponentFinder.java

```diff
@@ -132,7 +141,7 @@ private ComponentDto checkIsProject(ComponentDto component) {
   public OrganizationDto getOrganization(DbSession dbSession, ComponentDto component) {
     String organizationUuid = component.getOrganizationUuid();
     return dbClient.organizationDao().selectByUuid(dbSession, organizationUuid)
```

Comment:
Forget my "let's discusss", I found you were talking of new code. 
Totally agree to use  checkFoundWithOptional here